File: stack/BOJ/6198/6198.py
def main():   
    n = int(input())
    stack = []
    res = 0
    for _ in range(n):
        height = int(input())
        while (stack and stack[-1] <= height):
            stack.pop()
        res += len(stack)
        stack.append(height)
    print(res)

if __name__ == "__main__":
    main()

File: stack/BOJ/6198/6198.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
	stack<int> S;
	int N;
	
	long long res = 0; // Integer overflow 주의!!
	cin >> N;
	for (int i = 0; i < N; i++) {
		int num;
		cin >> num;
		while (!S.empty() && S.top() <= num) S.pop(); 
		res += S.size();
		S.push(num);
	}
	cout << res;
	return 0;
}
